### PR TITLE
Fix field registry typings

### PR DESCRIPTION
**Pull request by samelhusseini**
## The basics
- [x] I branched from develop
- [x] My pull request is against develop
- [x] My code follows the [style guide](https://developers.google.com/blockly/guides/modify/web/style-guide)

### Resolves

Fixes for https://github.com/google/blockly/issues/2937

### Proposed Changes

Typings fixes for field registry and some misc typings.

### Reason for Changes

Typescript typings.

### Test Coverage

Tested by running closure build and `npm run typings`.

Tested on:

### CODE REVIEW

File: core/interfaces/i_registrable_field.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+ */
+
+'use strict';
+
+goog.provide('Blockly.IRegistrableField');
+
+goog.requireType('Blockly.Field');
+
+/**
+ * A registrable field.
```
Comment (alschmiedt) [inline core/interfaces/i_registrable_field.js:19]: Add a comment somewhere in here explaining why we couldn't make it an interface.

File: core/options.js [hunk 2]
```diff
@@ -171,7 +173,7 @@ Blockly.Options = function(options) {
   /**
    * The parent of the current workspace, or null if there is no parent
    * workspace.
-   * @type {Blockly.Workspace}
+   * @type {Blockly.WorkspaceSvg}
```
Comment (alschmiedt) [inline core/options.js:176]: Maybe add a comment explaining why we can assume that this is a WorkspaceSvg.

**Review by alschmiedt — APPROVED** [review]

File: core/interfaces/i_registrable_field.js (hunk 1 above)
Comment (samelhusseini) [reply core/interfaces/i_registrable_field.js:19]: Done. Please check if that makes sense.

File: core/options.js (hunk 2 above)
Comment (samelhusseini) [reply core/options.js:177]: Done. Please check if that makes sense.

**Review by alschmiedt — APPROVED** [review]
Yup both make sense to me

**Pull request merged**